### PR TITLE
8316115: Parallel: Fix -Wconversion warnings around NUMA node ID

### PR DESCRIPTION
**Pull request by albertnetymk**
Essentially same as https://github.com/openjdk/jdk/pull/15541, but for Parallel.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8316115](https://bugs.openjdk.org/browse/JDK-8316115): Parallel: Fix -Wconversion warnings around NUMA node ID (**Enhancement** - P4)

### Reviewers
 * [Aleksey Shipilev](https://openjdk.org/census#shade) (@shipilev - **Reviewer**)
 * [Thomas Schatzl](https://openjdk.org/census#tschatzl) (@tschatzl - **Reviewer**) ⚠️ Review applies to [9d74efc9](https://git.openjdk.org/jdk/pull/15682/files/9d74efc90475e5ce2deec7b4a423215b05a798c6)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/15682/head:pull/15682` \
`$ git checkout pull/15682`

Update a local copy of the PR: \
`$ git checkout pull/15682` \
`$ git pull https://git.openjdk.org/jdk.git pull/15682/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 15682`

View PR using the GUI difftool: \
`$ git pr show -t 15682`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/15682.diff">https://git.openjdk.org/jdk/pull/15682.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/15682#issuecomment-1715584263)

### CODE REVIEW

**Review by shipilev — CHANGES_REQUESTED** [review]
See multiple failures in GHA :)

File: src/hotspot/share/gc/parallel/mutableNUMASpace.hpp [hunk 1]
```diff
@@ -117,7 +117,7 @@ class MutableNUMASpace : public MutableSpace {
       alloc_rate()->sample(alloc_rate_sample);
     }
 
-    int lgrp_id() const                             { return _lgrp_id;             }
+    uint lgrp_id() const                             { return _lgrp_id;             }
```
Comment (tschatzl) [inline src/hotspot/share/gc/parallel/mutableNUMASpace.hpp:120]: Indentation

**Review by tschatzl — APPROVED** [review]
Looks good. Please fix the indentation issue before pushing.

**Comment by albertnetymk** [comment]
@shipilev GHA builds are green now. The current test failures are unrelated to this PR.

**Review by shipilev — APPROVED** [review]

**Comment by albertnetymk** [comment]
Thanks for the review.

/integrate

**Pull request closed without merging**